src/test/java/com/truelayer/java/TestUtils.java

```diff
@@ -285,13 +283,46 @@ public static void runAndAssertHeadlessResourceAuthorisation(
         }
     }
 
+    public enum HeadlessResource {
+        PAYMENTS,
+        MANDATES
+    }
+
     @RequiredArgsConstructor
     @Getter
-    public enum HeadlessResourceAuthorization {
-        PAYMENTS("single-immediate-payments", "{\"action\":\"Execute\", \"redirect\": false}"),
-        MANDATES("vrp-consents", "{\"action\":\"Authorise\", \"redirect\": false}");
+    public enum HeadlessResourceAction {
+        EXECUTE("Execute"),
+        AUTHORISE("Authorise"),
+        REJECT_AUTHORISATION("RejectAuthorisation");
 
-        private final String path;
-        private final String payload;
+        private final String action;
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    public static class HeadlessResourceAuthorization {
```

Comment:
Refactored from enum to class with a builder to allow more flexibility, i.e. allow to select the `action` as well